src/API.jl

```diff
@@ -1273,30 +1273,41 @@ function precompile(ctx::Context, pkgs::Vector{PackageSpec}; internal_call::Bool
         end
     end
 
-    stderr_outputs = Dict{Base.PkgId,String}()
+    std_outputs = Dict{Base.PkgId,String}()
     taskwaiting = Set{Base.PkgId}()
     pkgspidlocked = Dict{Base.PkgId,String}()
 
-    function monitor_stderr(pkg, iob)
-        try
-            while isopen(iob)
-                str = readline(iob)
-                stderr_outputs[pkg] = get(stderr_outputs, pkg, "") * str * "\n"
-                if !in(pkg, taskwaiting) && occursin("waiting for IO to finish", str)
-                    !fancyprint && lock(print_lock) do
-                        println(io, pkg.name, color_string(" Waiting for background task / IO / timer.", Base.warn_color()))
+    function monitor_stderr_stdout(pkg, _stderr, _stdout)
+        @sync begin
+            @async try
+                while isopen(_stderr)
```

Comment:
`isopen` is the wrong predicate to use here, as stated in the docstring for it, use `eof`.